nixos/modules/services/desktops/deepin/deepin.nix

```diff
@@ -0,0 +1,101 @@
+# deepin
+
+{ config, pkgs, lib, ... }:
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.deepin.core.enable = lib.mkEnableOption "
+      Basic dbus and systemd services, groups and users needed by the
+      Deepin Desktop Environment.
+    ";
+
+    services.deepin.deepin-menu.enable = lib.mkEnableOption "
+      DBus service for unified menus in Deepin Desktop Environment.
+    ";
+
+    services.deepin.deepin-turbo.enable = lib.mkEnableOption "
+      Turbo service for the Deepin Desktop Environment. It is a daemon
+      that helps to launch applications faster.
+    ";
+
+  };
+
+
+  ###### implementation
+
+  config = lib.mkMerge [
+
+    (lib.mkIf config.services.deepin.core.enable {
+      environment.systemPackages = [
+        pkgs.deepin.dde-api
+        pkgs.deepin.dde-calendar
+        pkgs.deepin.dde-daemon
+        pkgs.deepin.dde-session-ui
+        pkgs.deepin.deepin-image-viewer
+        pkgs.deepin.deepin-screenshot
+      ];
+
+      services.dbus.packages = [
+        pkgs.deepin.dde-api
+        pkgs.deepin.dde-calendar
+        pkgs.deepin.dde-daemon
+        pkgs.deepin.dde-session-ui
+        pkgs.deepin.deepin-image-viewer
+        pkgs.deepin.deepin-screenshot
+      ];
+
+      systemd.packages = [
+        pkgs.deepin.dde-api
+        pkgs.deepin.dde-daemon
+      ];
+
+      users.groups.deepin-sound-player = { };
+
+      users.users.deepin-sound-player = {
+        description = "Deepin sound player";
+        group = "deepin-sound-player";
+        isSystemUser = true;
+      };
+
+      users.groups.deepin-daemon = { };
+
+      users.users.deepin-daemon = {
+        description = "Deepin daemon user";
+        group = "deepin-daemon";
+        isSystemUser = true;
+      };
+
+      security.pam.services.deepin-auth-keyboard.text = ''
+        # original at ${pkgs.deepin.dde-daemon}/etc/pam.d/deepin-auth-keyboard
```

Comment:
I'm not familiar with pam in general, but I do know that you'll also need to adapt
the config at `share/pam-configs/deepin-auth `.

Reply:
I'm also not familiar with pam. The file path has already been fixed. What else would have to be adapted in the config at `share/pam-configs/deepin-auth`?

```
Name: Deepin authentication
Default: yes
Priority: 500
Conflicts:
Auth-Type: Primary
Auth:
	[success=end default=ignore]	pam_exec.so quiet seteuid expose_authtok /nix/store/nank2x64fl43x1vkzik3xq6ck0b35hbn-dde-daemon-3.24.1/lib/deepin-daemon/dde-authority-pam
```

Reply:
I did not found any similar case in nixpkgs. There is only one mention of `pam-configs` (in `./pkgs/tools/security/chaps/default.nix`), but it is unrelated.

Reply:
`pam-configs` are used for [pam-auth-update](https://manpages.ubuntu.com/manpages/disco/man8/pam-auth-update.8.html)

> is  a  utility that permits configuring the central authentication policy
       for the system using pre-defined profiles as supplied by PAM  module  packages.   Profiles
       shipped  in  the  /usr/share/pam-configs/  directory specify the modules, with options, to
       enable; the preferred ordering with respect to  other  profiles;  and  whether  a  profile
       should  be  enabled by default.  Packages providing PAM modules register their profiles at
       install time by calling pam-auth-update --package.

Obviously something that would have just worked as expected for debian based distros, but very useless here. 

I guess it would end up like this for use in [`modules/security/pam.nix`](https://github.com/NixOS/nixpkgs/blob/4616b4ec85cd1779cda1dc8519288aebb1c4011d/nixos/modules/security/pam.nix)
```
auth [success=end default=ignore] pam_exec.so quiet seteuid expose_authtok ${deepin.dde-daemon}/lib/deepin-daemon/dde-authority-pam

Reply:
@worldofpeace I think it is better to remove the commit dealing with PAM from this PR and create a new PR for it after this one is merged.

Reply:
Yep, that's what I meant in https://github.com/NixOS/nixpkgs/pull/59943#pullrequestreview-233206194